src/Cake.Common/Build/GitHubActions/Commands/GitHubActionsCommands.cs

```diff
@@ -52,7 +45,7 @@ public GitHubActionsCommands(
             _fileSystem = fileSystem ?? throw new ArgumentNullException(nameof(fileSystem));
             _writer = writer ?? throw new ArgumentNullException(nameof(writer));
             _actionsEnvironment = actionsEnvironment ?? throw new ArgumentNullException(nameof(actionsEnvironment));
-            _createHttpClient = createHttpClient ?? throw new ArgumentNullException(nameof(createHttpClient));
+            _artifactsService = new GitHubActionsArtifactService(environment, fileSystem, actionsEnvironment, createHttpClient ?? throw new ArgumentNullException(nameof(createHttpClient)));
```

Comment:
Is there a reason that this isn't being injected into the constructor?

Reply:
I guess it could, it was private methods in same class I just refactored out to an internal resource to be able easier test/debug in isolation, while keeping the same public interface of GitHubActionsCommands.
But could try to refactor and see if the internal can be injected without issue,

Reply:
> @devlead said...
it was private methods in same class I just refactored out to an internal resource to be able easier test/debug in isolation, while keeping the same public interface of GitHubActionsCommands.

Ah gotcha!  I see what you mean. I think keeping the API the same makes sense, and the abstraction of the private methods into a service class makes sense.

I don't think that there is anything to do here, other than perhaps leave a comment in the code for future us, to remember why this service wasn't injected via the constructor.

Reply:
```suggestion
            // Internal service class, keeping public API unchanged. 
            _artifactsService = new GitHubActionsArtifactService(environment, fileSystem, actionsEnvironment, createHttpClient ?? throw new ArgumentNullException(nameof(createHttpClient)));
```

Something like this?

Reply:
Yeah, I think that looks good. I might add something like ", introduced in PR https://...". Just so our future selves can find it easily.

Reply:
Added